### DISCUSSION TITLE
[Request additional features]Reduce the number and frequency of remote API calls

### DISCUSSION THREAD

**Discussion by yoyofreeman** [RainbowMiner/RainbowMiner#2650] [Ideas]
Let RBMclient subscribe to API messages from RBMserver instead of querying the target directly.
I run an RBM cluster with about 30 clients (workers). They have just been built. Recently, I often encounter the problem of being temporarily blocked by the remote API service, which leads to unreasonable and wrong switching of RBM. So I request to add this feature.

**Comment by yoyofreeman** [comment]
I conducted some analysis and experiments with some technical experts on this, and it is likely that cloudflare is blocking API requests.

**Reply by RainbowMiner** [reply]
Are you using a server/client structure, or are all those clients running standalone?

**Reply by yoyofreeman** [reply]
I‘m using server/client structure. And they are all in the same network segment.
And they are all in the same network segment. 1 unit uses server mode and 28 units use client mode.

**Reply by yoyofreeman** [reply]
My internet bandwidth is very limited and unreliable (due to some insurmountable limitation, my miners have to connect to the internet via VPN), and every time RBM upgrades and downloads a new version of the miner, each of my miners does the same thing. This is neither wise nor necessary, so it is recommended that file downloads and API data query operations in server/client mode be completed based on server-side file and data caching.

**Reply by RainbowMiner** [reply]
Thank you, this is a very good idea. I'll try to figure something out.
